Gopro-Sync/goprosync.py

```diff
@@ -0,0 +1,159 @@
+    shutil.rmtree(curr_directory)
+
+    # Continue off from before, creating channels and times 
+
+    times_left = np.linspace(0, left_samples/left_freq, num=left_samples)
+    times_right = np.linspace(0, right_samples/right_freq, num=right_samples)
+
+    # Use right channel for left camera and left channel for right camera
+    left_channel = left_signal_array[1::2]
+    right_channel = right_signal_array[0::2]
+
+    # Find the impulse. The impulse should be the highest value in the signal.
+    # Use the region around the impulse for matrix profile later
+
+    left_spike = left_channel.argmax()
+    right_spike = right_channel.argmax()
+
+    left_spike_start = left_spike-10000
+    left_spike_end = left_spike+10000
+
+    right_spike_start = right_spike-10000
+    right_spike_end = right_spike+10000
+
+    # Prepare the subarrays for the matrix profile algorithm
+    # The matrix profile algorithm, specifically for conserved pattern detection
+    # To read more about matrix profiles, visit matrixprofile.org
+
+    m = 10000
+    left_subarray = left_channel[left_spike_start:left_spike_end]
+    right_subarray = right_channel[right_spike_start:right_spike_end]
+
+    # Matrix profile can find patterns in nlogn time, 
+    # so we use it to quickly find conserved patterns.
+    matrix_profile=stumpy.stump(T_A=left_subarray.astype(np.float64),
+                                  m=m, T_B=right_subarray.astype(np.float64), ignore_trivial=False)
+
+    # Find the index of the left and right motifs
+    left_motif_index = matrix_profile[:,0].argmin()
+    right_motif_index = matrix_profile[left_motif_index,1] + right_spike_start
+    left_motif_index += left_spike_start    
+
+    # Extract the desired cropped left camera video
+    left_video = VideoFileClip(left.as_posix())
+    left_video_trimmed = left_video.subclip(times_left[left_motif_index],times_left[-1])
+    left_video_trimmed.write_videofile(trimmed_left.as_posix())
+    left_video.close()
+    left_video_trimmed.close()
+
+    right_video = VideoFileClip(right.as_posix())
+    right_video_trimmed = right_video.subclip(times_right[right_motif_index],times_right[-1])
+    right_video_trimmed.write_videofile(trimmed_right.as_posix())
+    right_video.close()
+    right_video_trimmed.close()
+
+
+def main():
+    """Check arguments and set up paths, then call goprosync function
+    """
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument('--left', type=Path, required=True)
+    parser.add_argument('--right', type=Path, required=True)
+    parser.add_argument('--out', type=Path, required=True)
+
+    args = parser.parse_args()
+
+    # Make sure that all the paths are valid
+
+    if not args.left.is_file():
+        print("Left path is not a file.")
+        sys.exit(1)
+
+    elif not args.right.is_file():
+        print("Right path is not file")
+        sys.exit(1)
+
+    # Create output directory or use provided one if exists
+
+    args.out.mkdir(exist_ok=True)
```

Comment:
What if the parents of this directory don't exist?  Recommend also including `parents=True`